### PR TITLE
OOP approach for Text frame handling.

### PR DESCRIPTION
**Pull request by MadCat9k**

### CODE REVIEW

**Comment by alesker** [comment]
Seems good to me. Although it's a shame we can't make SRTextFrame a protocol. :shipit: 
P.S. In Swift one can define `init` in protocol scope. 🙄

**Comment by nspavlo** [comment]
Why `SRNoICUFrameFactory` inherits from `SRFrameFactory` if it is pure `SRFrameFactoryProtocol` implementation? `SRNoICUFrameFactory` is a `SRFrameFactory`? Am I missing something?

**Comment by MadCat9k** [comment]
Yeah, my bad. It should be a NSObject conforming to  SRFrameFactoryProtocol. Will fix this ASAP.

File: SocketRocket/SRTextFrame.m [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+- (NSString *)value
+{
+    return [self valueWithLength:self.data.length];
+}
+
+- (int32_t)length
+{
+    if ([self.data length] > INT32_MAX) {
```
Comment (nspavlo) [inline SocketRocket/SRTextFrame.m]: NSData length is declared as property. Dot notation might be better approach.

File: SocketRocket/FrameFactory/SRFrameFactory.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//  Created by Kirils Sivokozs on 24/04/2017.
+//
+//
+
+#import "SRFrameFactoryProtocol.h"
+#import <Foundation/Foundation.h>
+
+@class SRTextFrame;
+
+@interface SRFrameFactory : NSObject <SRFrameFactoryProtocol>
```
Comment (nspavlo) [inline SocketRocket/FrameFactory/SRFrameFactory.h]: At this point of time is quite hard to evaluate both factories. creating objects directly will be simpler and will introduce same count of new dependencies. For me it violates YAGNI principle.

**Comment by MadCat9k** [comment]
@alesker @nspavlo I threw out pointless factory. Feel free to comment.

**Comment by alesker** [comment]
:shipit:

**Pull request merged**